### PR TITLE
[tween] on/off node events in action-manager.ts only when target node is valid.

### PR DESCRIPTION
**Pull request by dumganhar**
Found a bug in https://github.com/cocos/cocos-engine/pull/17141 , fix it here and add an unit test for it.

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: tests/tween/tween.test.ts [hunk 1]
```diff
@@ -4202,5 +4202,39 @@ test('tween custom object, ensure same type', function () {
...
+    director.registerSystem(TweenSystem.ID, sys, System.Priority.MEDIUM);
+
+    const node = new Node();
+
+    tween(node).to(1, { position: new Vec3(100, 100, 100) }).start();
+
+    runFrames(1);// Start
+
+    runFrames(30);
+    expect(node.position.equals(v3(50, 50, 50))).toBeTruthy();
```
Comment (minggo) [inline tests/tween/tween.test.ts:4476]: Why need to test it in this unit test?

Comment (dumganhar) [reply tests/tween/tween.test.ts:4476]: Just to ensure the tween is running some frames.

File: tests/tween/tween.test.ts [hunk 2]
```diff
@@ -4202,5 +4202,39 @@ test('tween custom object, ensure same type', function () {
...
+
+    node._destroyImmediate();
+
+    expect(node.position).toBeNull();
+    expect(node.scale).toBeNull();
+
+    Tween.stopAllByTarget(node);
```
Comment (dumganhar) [inline tests/tween/tween.test.ts:4483]: To ensure there are not any script errors while invoking stopAllByTarget. If there are errors, the unit test will fail.

File: tests/tween/tween.test.ts [hunk 3]
```diff
@@ -4202,5 +4202,39 @@ test('tween custom object, ensure same type', function () {
...
+
+    runFrames(30);
+
+    // node was destroyed, it should not be updated any more.
+    tween(node).to(1, { scale: new Vec3(3, 3, 3) }).start();
+
+    runFrames(30);
+
+    expect(node.position).toBeNull();
+    expect(node.scale).toBeNull();
```
Comment (dumganhar) [inline tests/tween/tween.test.ts:4493]: The scale should not be changed since node was destroyed.

**Review by minggo — APPROVED** [review]

**Pull request merged**